quinn/src/connection.rs

```diff
@@ -730,31 +737,35 @@ impl ConnectionRef {
         udp_state: Arc<UdpState>,
         runtime: Arc<dyn Runtime>,
     ) -> Self {
-        Self(Arc::new(Mutex::new(ConnectionInner {
-            inner: conn,
-            driver: None,
-            handle,
-            on_handshake_data: Some(on_handshake_data),
-            on_connected: Some(on_connected),
-            connected: false,
-            timer: None,
-            timer_deadline: None,
-            conn_events,
-            endpoint_events,
-            blocked_writers: FxHashMap::default(),
-            blocked_readers: FxHashMap::default(),
-            stream_opening: [Arc::new(Notify::new()), Arc::new(Notify::new())],
-            incoming_uni_streams_reader: None,
-            incoming_bi_streams_reader: None,
-            datagram_reader: None,
-            finishing: FxHashMap::default(),
-            stopped: FxHashMap::default(),
-            closed: Arc::new(Notify::new()),
-            error: None,
-            ref_count: 0,
-            udp_state,
-            runtime,
-        })))
+        Self(Arc::new(ConnectionInner {
+            state: Mutex::new(State {
```

Comment:
Maybe `State` or `ConnectionInner` could do with an associated `new()` function (as a separate change), what do you think?

Reply:
I'd rather not increase the number of places that the eight arguments here get passed through, and which must be updated for any change to them. Did realize we can `Default` `Shared` for now, though!